tests/ttnn/unit_tests/operations/ccl/test_all_gather.py

```diff
@@ -1929,3 +1951,132 @@ def test_all_gather_fp32(  # https://github.com/tenstorrent/tt-metal/issues/9686
         tt_output_tensor = t.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch()
         eq, output = comp_equal(tt_output_tensor, input_tensor)
         assert eq, f"{i} FAILED: {output}"
+
+
+@skip_for_grayskull("Requires eth connected devices to run")
+@pytest.mark.parametrize("num_devices", [8])
+@pytest.mark.parametrize("dim", [3])
+@pytest.mark.parametrize("tensor_layout", [ttnn.TILE_LAYOUT])
+@pytest.mark.parametrize(
+    "input_dtype",
+    [
+        ttnn.bfloat16,
+    ],
+)
+@pytest.mark.parametrize(
+    "tensor_mem_layout",
+    [
+        ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+        ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
+        ttnn.TensorMemoryLayout.BLOCK_SHARDED,
+    ],
+)
+@pytest.mark.parametrize("orientation", [ttnn.ShardOrientation.ROW_MAJOR])
+@pytest.mark.parametrize("num_links", [1])
+@pytest.mark.parametrize(
+    "input_shape, input_shard_shape,shard_grid",
+    (
+        # LLama
+        (
+            (4, 1, 256, 32),
+            (32, 32),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(7, 3))}),
+        ),
+    ),
+)
+@pytest.mark.parametrize("tile_h", [2])
+@pytest.mark.parametrize("enable_async", [True])
+def test_tiny_all_gather_sharded_post_commit(
+    t3k_mesh_device,
+    num_devices,
+    input_shape,
+    input_shard_shape,
+    shard_grid,
+    dim,
+    num_links,
+    orientation,
+    input_dtype,
+    tensor_layout,
+    tensor_mem_layout,
+    # num_cores,
+    use_program_cache,
+    function_level_defaults,
+    enable_async,
+    tile_h,
+):
+    run_all_gather_sharded_t3k(
+        t3k_mesh_device,
+        num_devices,
+        input_shape,
+        input_shard_shape,
+        shard_grid,
+        dim,
+        num_links,
+        orientation,
+        input_dtype,
+        tensor_layout,
+        tensor_mem_layout,
+        # num_cores,
+        use_program_cache,
+        function_level_defaults,
+        all_gather_topology=ttnn.Topology.Ring,
+        enable_async=enable_async,
+        tile=(tile_h, 32),
+    )
+
+
+@skip_for_grayskull("Requires eth connected devices to run")
+@pytest.mark.parametrize(
+    "num_devices, num_links, input_shape, dim, layout",
+    [
+        (4, 1, [1, 1, 32, 16384], 3, ttnn.TILE_LAYOUT),
```

Comment:
is it possible to test dimensions which are not divisible by tile size?

Reply:
This is a limitation in the Tensor API that converts torch or row-major tensors into tiled tensors, the shape height and width must be a multiple of the tile height and width. See ttnn/cpp/ttnn/tensor/tensor_impl.hpp line 209. If there is a use case where we want shapes that are not multiples of the tile sizes in tile configuration, an issue should be raised to support any arbitrary shape size. I tested [1, 1, 16, 16384] and [1, 1, 48, 16384]  which works for the tiny tile configurations but as expected fails for (32,32) full tiles, however I think we should only include shapes that work universally in the test plan.